app/src/main/java/edu/uc/haflga/financialpal/dto/Category.kt

```diff
@@ -6,7 +6,7 @@ class Category(var name: String, var limit: Limit?) {
 
     var id: Int = 0
     //Creating an ArrayList for transaction Categories
-    var allCategories = ArrayList<Category>()
+    private var allCategories = ArrayList<Category>()
```

Comment:
Added private tag to arrayList: https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules

Reply:
Good... in this case, private is a more appropriate modifier than the default, which is public.